src/modules/roc_audio/frame.h

```diff
@@ -20,9 +20,43 @@ namespace roc {
 namespace audio {
 
 //! Audio frame.
-struct Frame {
-    //! Frame samples.
-    core::Slice<sample_t> samples;
+class Frame {
+public:
+    //! Construct empty frame.
+    Frame();
+
+    //! Construct frame from samples.
+    Frame(const core::Slice<sample_t>& samples);
+
+    //! Frame flags.
+    enum {
+        //! Set if no packets were extracted to the frame when the frame was built.
+        FlagEmpty = (1 << 0),
+
+        //! Set if some queued packets were regarded as outdated and dropped when the
+        //! frame was built.
+        FlagSkip = (1 << 1)
+    };
+
+    //! Add flags.
+    void add_flags(unsigned flags);
+
+    //! Returns true if frame has no packets.
+    bool is_empty() const;
```

Comment:
I'd rather add a single flags() method instead of adding a getter method for every flag. The advantage of the first approach are:

* it's easier to grep all occurrences of a flag in the source code
* it's clear that we just check a flag that was set previously with add_flags() and no other magic happens

Reply:
Fixed  in e5282d5358d4a150f3b393acd3eded5ade763ef5